### PR TITLE
feature/Api

### PR DESCRIPTION
**Pull request by Regiss05**
add api in apiActions.js

### CODE REVIEW

File: src/redux/books/apiActions.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    data: {
+      item_id: book.id,
+      title: book.title,
+      author: book.author,
+      category: 'abc',
+    },
+  }).catch(() => {
+    dispatch(fetchError());
+  });
+  dispatch(getBookApi());
```
Comment (AnselemOdims) [inline src/redux/books/apiActions.js:36]: - [x] Please kindly ensure that the addition of book works as it should be. At the moment, when you add a book, this is the result in the browser.
- [x] When you refresh the browser the added book disappears. Please ensure that the API call works as it should.
- [x] Also, one can only add one book. When you try to add again, it does not work.

image

File: src/redux/books/apiActions.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  });
+};
+
+export const removeBookApi = (ID) => (dispatch) => {
+  baseUrl.delete(`/books/${ID}`, { item_id: ID }).then(() => {
+    dispatch(getBookApi());
+  }).catch(() => {
+    dispatch(fetchError());
+  });
+};
```
Comment (AnselemOdims) [inline src/redux/books/apiActions.js:21]: - [x] Please also ensure that the removing book from the API works as expected.

File: src/redux/books/apiActions.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps';
+const appId = 'ux6EkcA2ZXp2cRIPwiAL';
+const baseUrl = axios.create({ baseURL: `${url}/${appId}` });
+export const getBookApi = () => (dispatch) => {
+  baseUrl.get('/books').then((res) => {
+    dispatch(addBook(res.data));
+  }).catch(() => {
+    dispatch(fetchError());
+  });
+};
```
Comment (AnselemOdims) [inline src/redux/books/apiActions.js:13]: - [x] At the moment, when you reload the browser the books are not fetched from the API. Please kindly ensure the fetching of books works as expected.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @Regiss05 , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
- [x] You can add the `redux-logger` library so that you can see the logs of the API interaction in the browser console.

_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Lameck1 — APPROVED** [review]
Hi, @Regiss05,

Well done working on the previous review concerns 👏🏼.
Tasks for this project are now successfully implemented 💯.

### Highlights
- Clean code ✔️ 
- Good commit messages ✔️ 

All the best with the next project.

You are free to merge this branch :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**